File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_05_navigation/ui/NavigationMainActivity.java


package com.kunminx.jetpack_java.sample_05_navigation.ui;

import android.os.Bundle;

import androidx.databinding.DataBindingUtil;

import com.kunminx.architecture.ui.BaseActivity;
import com.kunminx.jetpack_java.R;
import com.kunminx.jetpack_java.sample_05_navigation.ui.callback.SharedViewModel;

/**
 * Create by KunMinX at 19/10/16
 */

public class NavigationMainActivity extends BaseActivity {

    private SharedViewModel mSharedViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mSharedViewModel = getActivityViewModel(SharedViewModel.class);

        DataBindingUtil.setContentView(this, R.layout.activity_main_navigation);

        mSharedViewModel.closeActivity.observe(this, aBoolean -> {
            finish();
        });
    }

}

File: jetpack_kotlin/src/main/java/com/flywith24/jetpack_kotlin/sample_05_navigation/ui/NavigationMainActivity.kt
package com.flywith24.jetpack_kotlin.sample_05_navigation.ui

import android.os.Bundle
import androidx.activity.viewModels
import com.flywith24.jetpack_kotlin.R
import com.flywith24.jetpack_kotlin.sample_05_navigation.ui.callback.SharedViewModel
import com.kunminx.architecture.kotlin.observeEvent
import com.kunminx.architecture.ui.BaseActivity

/**
 * @author Flywith24
 * @date   2020/6/1
 * time   21:56
 * description
 */
class NavigationMainActivity : BaseActivity(R.layout.kotlin_activity_main_navigation) {
    private val mSharedViewModel by viewModels<SharedViewModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mSharedViewModel.closeActivity.observeEvent(this) {
            finish()
        }
    }

}